packages/x-date-pickers/src/PickersTextField/PickersInputBase/pickersInputBaseClasses.ts

```diff
@@ -12,6 +12,8 @@ export interface PickersInputBaseClasses {
   readOnly: string;
   /** State class applied to the root element if `error=true`. */
   error: string;
+  /** State class applied to the root element if `size=small`. */
```

Comment:
I didn’t add @deprecated here because, in the @mui/material TextField component, inputSizeSmall is deprecated due to the presence of both `input` and `sizeSmall`  (https://mui.com/material-ui/api/input-base/#input-base-classes-MuiInputBase-sizeSmall) classes. However, in PickersInputBase, the sizeSmall  class isn’t available, so it’s not possible to compose the sizeSmall class in this context.

Additionally we can add `sizeSmall` class and then deprecate `inputSizeSmall` and make it consistent with @mui/material TextField. Let me know, if we want to take that approach